### PR TITLE
fix: Added element IDs for automation

### PR DESCRIPTION
**Pull request by omerhabib26**
### Description:

- Added Element IDs for automation to Views

### CODE REVIEW

**Review by farhan-arshad-dev — CHANGES_REQUESTED** [review]

File: profile/src/main/java/org/openedx/profile/presentation/edit/EditProfileFragment.kt [hunk 1]
```diff
@@ -162,21 +162,6 @@ class EditProfileFragment : Fragment() {
...
-                override fun handleOnBackPressed() {
-                    if (viewModel.profileDataChanged) {
-                        viewModel.setShowLeaveDialog(true)
-                    } else {
-                        requireActivity().supportFragmentManager.popBackStack()
-                    }
-                }
-            })
-    }
-
```
Comment (farhan-arshad-dev) [inline profile/src/main/java/org/openedx/profile/presentation/edit/EditProfileFragment.kt:176]: why need this code deletion?

File: core/src/main/java/org/openedx/core/extension/StringExt.kt [hunk 2]
```diff
@@ -29,4 +30,6 @@ fun String.replaceLinkTags(isDarkTheme: Boolean): String {
     return text
 }
 
-fun String.withoutSpaces(): String = this.replace(" ", "")
+fun String.replaceSpaces(target: String = ""): String = this.replace(" ", target)
```
Comment (farhan-arshad-dev) [inline core/src/main/java/org/openedx/core/extension/StringExt.kt:33]: better to remove it with `replaceSpace`

**Review by farhan-arshad-dev — APPROVED** [review]
just some nits.

File: profile/src/main/java/org/openedx/profile/presentation/edit/EditProfileFragment.kt (hunk 1 above)
Comment (omerhabib26) [reply profile/src/main/java/org/openedx/profile/presentation/edit/EditProfileFragment.kt:176]: it was unused

Comment (omerhabib26) [reply profile/src/main/java/org/openedx/profile/presentation/edit/EditProfileFragment.kt:176]: it was unused

**Review by farhan-arshad-dev — APPROVED** [review]

**Review by farhan-arshad-dev — APPROVED** [review]

**Pull request merged**